### ISSUE TITLE
Missed substitution when class and builder have generics

### ISSUE THREAD

**Issue by akozlova** [mplushnikov/lombok-intellij-plugin#313]
Given simplified example from https://youtrack.jetbrains.com/issue/IDEA-164687
```
@Data
@Builder
public class ObjectApiResponse<K> {
	private K response;
	public static <Z> void create(Z res, ObjectApiResponseBuilder<Z> builder) {
		ObjectApiResponseBuilder<Z> response1 = builder.response(res);
	}

	public static class ObjectApiResponseBuilder<T> {}
}
```
Lombok generates light method 
``` ObjectApiResponseBuilder<T> response(K response)```
which then leads to problems during highlighting, refactorings, etc

**Comment by mplushnikov** [comment]
Similar problem #306

**Comment by akozlova** [comment]
Just described how to fix the issue here https://intellij-support.jetbrains.com/hc/en-us/community/posts/206006504-Psi-Type-for-generics?page=1#community_comment_205126304

**Comment by AlexejK** [comment]
Should be fixed by #329

**Comment by mplushnikov** [comment]
Great help @akozlova and @AlexejK ! Thank you very much for this contribution!
